crates/syntax/src/ast/make.rs

```diff
@@ -196,9 +204,9 @@ pub fn path_from_segments(
 ) -> ast::Path {
     let segments = segments.into_iter().map(|it| it.syntax().clone()).join("::");
     ast_from_text(&if is_abs {
-        format!("use ::{};", segments)
+        format!("fn f(x: ::{}) {{}}", segments)
```

Comment:
I need to change this to preserve `<T as Trait>` type of paths. Is this fine? Shouldn't all path-related constructors this approach instead of `use {}`?

Reply:
Ye this change is good, otherwise you could easily panic by feeding non use-tree path segments in there.